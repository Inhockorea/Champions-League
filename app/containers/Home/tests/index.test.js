/**
 * Test the Home
 */

import React from 'react';
import { render } from 'react-testing-library';
import { Provider } from 'react-redux';
import { browserHistory } from 'react-router-dom';

import { Home, mapDispatchToProps } from '../index';
import configureStore from '../../../configureStore';

describe('<Home />', () => {
  let store;

  beforeAll(() => {
    store = configureStore({}, browserHistory);
  });

  it('should render and match the snapshot', () => {
    const {
      container: { firstChild },
    } = render(
      <Provider store={store}>
        <Home
          teamNames={['Arsenal']}
          loading={false}
          error={false}
          dispatchloadTeamNames={jest.fn()}
        />
      </Provider>,
    );
    expect(firstChild).toMatchSnapshot();
  });

  it('should fetch the teamNames on mount if error is false', () => {
    const mockFetchingTeamNames = jest.fn();
    render(
      <Provider store={store}>
        <Home
          teamNames={['Arsenal']}
          loading={false}
          error={false}
          dispatchloadTeamNames={mockFetchingTeamNames}
        />
      </Provider>,
    );
    expect(mockFetchingTeamNames).toHaveBeenCalled();
  });

  describe('mapDispatchToProps', () => {
    describe('dispatchloadTeamNames', () => {
      it('should be injected', () => {
        const dispatch = jest.fn();
        const result = mapDispatchToProps(dispatch);
        expect(result.dispatchloadTeamNames).toBeDefined();
      });
    });
  });
});
