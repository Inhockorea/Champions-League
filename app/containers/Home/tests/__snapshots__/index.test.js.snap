// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Home /> should render and match the snapshot 1`] = `
<section
  class="Section-eaizcf-0 lnleCN"
>
  <h1
    class="H1-tncdg6-0 jcnLlZ"
  >
    Champions League Participants
  </h1>
  <div
    class="Wrapper-sc-1umgotm-0 laCwDT"
  >
    <ul
      class="Ul-mwnq6h-0 jcxXFy"
    >
      <li
        class="Wrapper-euo0oy-0 jLvxtc"
      >
        <div
          class="Item-sc-3y9mie-0 fKXSkT"
        >
          Arsenal
        </div>
      </li>
    </ul>
  </div>
</section>
`;
